Challenges/challenge_46_can_place_flowers/greghilston/src/main/java/Solution.java

```diff
@@ -0,0 +1,54 @@
+public class Solution  {
+    /**
+     * Determines if we can plant a flower at a given location in a specific flowerbed.
+     * @param flowerBed state of flowerbed
+     * @param i location to check
+     * @return whether we can plant a flower or not
+     */
+    private boolean canPlantHere(int[] flowerBed, int i) {
+        // left check
+        int leftNeighborIndex = i - 1;
+
+        if(leftNeighborIndex > 0) // ensures we don't go off the flower bed
+        {
+            if(flowerBed[leftNeighborIndex] == 1) {
+                return false; // Can't plant next to a flower
+            }
+        }
+
+        // right check
```

Comment:
Code duplication from here on.

Reply:
Can you elaborate on this? Aren't checking the right of us now, so this is infact like unique code?

Unless your suggesting I factor out the actual logic into a little helper function and pust pass in the left/right along with the edge condition check?

Reply:
Yeah, that's what I mean. Code is redundant when it does the same thing structurally as another piece of code, even if there's differences to the specifics. These differences can be implemented into helper methods with parameters. Although in this case, in comparison, it wouldn't even hurt to just naively check both bounds at all times.